src/cljstyle/format/core.clj

```diff
@@ -179,7 +198,10 @@
 
       (and (get-in rules-config [:eof-newline :enabled?])
            (not (str/ends-with? (:formatted result) "\n")))
-      (update :formatted str "\n"))))
+      (update :formatted str "\n")
+
+      (not (get-in rules-config [:eof-newline :trailing-blanks?]))
+      (update :formatted trim-trailing-newlines rules-config))))
```

Comment:
Per the other comment, this should not apply changes if the rule isn't enabled.

Reply:
Curious about style for this. Should I have nested `cond->`s or redundant condition checks?

Nested `cond->`s:
```
(cond-> result
  shebang
  (update :formatted (partial str shebang))

  (get-in rules-config [:eof-newline :enabled?])
  (cond->
    (not (str/ends-with? (:formatted result) "\n"))
    (update :formatted str "\n")

    (not (get-in rules-config [:eof-newline :trailing-blanks?]))
    (update :formatted trim-trailing-newlines rules-config)))
```


Redundant condition checks:
```
(cond-> result
  shebang
  (update :formatted (partial str shebang))

  (and (get-in rules-config [:eof-newline :enabled?])
       (not (str/ends-with? (:formatted result) "\n")))
  (update :formatted str "\n")

  (and (get-in rules-config [:eof-newline :enabled?])
       (not (get-in rules-config [:eof-newline :trailing-blanks?])))
  (update :formatted trim-trailing-newlines rules-config))))

```